include/tbb/tbb_machine.h

```diff
@@ -918,7 +924,11 @@ inline bool __TBB_TryLockByte( __TBB_atomic_flag &flag ) {
 #ifndef __TBB_LockByte
 inline __TBB_Flag __TBB_LockByte( __TBB_atomic_flag& flag ) {
     tbb::internal::atomic_backoff backoff;
-    while( !__TBB_TryLockByte(flag) ) backoff.pause();
+    do {
+        while( __TBB_IsLocked(flag) )
+            backoff.pause();
+    }
+    while( !__TBB_TryLockByte(flag) ) ;
```

Comment:
```suggestion
    while (__TBB_IsLocked(flag) ||  !__TBB_TryLockByte(flag)) {
        backoff.pause();
    }
```

Reply:
Like this suggestion, simpler to read.  Slight behavior difference that after a !__TBB_TryLockByte(flag) pause will be immediately called vs checking __TBB_IsLocked(flag)  again, but I think that might be an improvement.